src/overlays/actors/ovl_En_Ex_Ruppy/z_en_ex_ruppy.c

```diff
@@ -21,29 +34,345 @@ const ActorInit En_Ex_Ruppy_InitVars = {
+        this->actionFunc = func_80A0AEE0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AA3C.s")
+void func_80A0AD88(EnExRuppy *this, GlobalContext *globalCtx) {
+    EnDivingGame* divingGame;
+    Vec3f unusedVec3f;
+    Vec3f unusedVec3f2;
+    const f32 localConst = 30.0f;
+    unusedVec3f = D_80A0B388;
+    unusedVec3f2 = D_80A0B394;
+    if (this->timer == 0) {
+        this->timer = 10;
+        func_800293E4(globalCtx, &this->actor.posRot.pos, 0.0f, 5.0f, 5.0f, Math_Rand_ZeroFloat(0.03f) + 0.07f);
+    }
+    if (DIVING_GAME != NULL) {
+        divingGame = DIVING_GAME;
+        if (divingGame->actor.update != NULL) {
+            if (divingGame->unk_29C == 0) {
+                this->timer = 20;
+                this->actionFunc = func_80A0AEE0;
+                return;
+            }
+            if (this->actor.xyzDistFromLinkSq < SQ(localConst)) {
+                Rupees_ChangeBy(this->value);
+                func_80078884(NA_SE_SY_GET_RUPY );
+                divingGame->unk_2A4++;
+                Actor_Kill(&this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AB1C.s")
+            }
+        } else {
+            Actor_Kill(&this->actor);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AC88.s")
+void func_80A0AEE0(EnExRuppy *this, GlobalContext *globalCtx) {
+    this->isFalling = this->isFalling + 1;
+    this->isFalling = this->isFalling & 1;
+    if (this->timer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AD88.s")
+void EnExRuppy_BlowUp(EnExRuppy *this, GlobalContext *globalCtx) {
+    f32 distToBlowUp;
+    Vec3f tempPoint1Vec;
+    Vec3f tempZeroVector;
+    s16 tempExplosionScale;
+    s16 tempExplosionScaleStep;
+    tempPoint1Vec = sPoint1Vec;
+    tempZeroVector = sZeroVector;
+    distToBlowUp = 50.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AEE0.s")
+    if (this->unk_152 == 2) {
+        distToBlowUp = 30.0f;
+    }
+    if (this->actor.xyzDistFromLinkSq < SQ(distToBlowUp)) {
+        if (this->actor.attachedA != 0) {
+            if (this->actor.attachedA->update != 0) {
+                DIVING_GAME->transitionDrawTable[15].z = (u16)1;
+            }
+        } else {
+            //That idiot! error
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ そ、そんなばかな！エラー！！！！！ ☆☆☆☆☆ \n" VT_RST);
+        }
+        //Stupid!
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ バカめ！ ☆☆☆☆☆ \n" VT_RST);
+        tempExplosionScale = 100;
+        tempExplosionScaleStep = 30;
+        if (this->unk_152 == 2) {
+            tempExplosionScale = 20;
+            tempExplosionScaleStep = 6;
+        }
+        EffectSsBomb2_SpawnLayered(globalCtx, &this->actor.posRot.pos, &tempZeroVector, &tempPoint1Vec, tempExplosionScale, tempExplosionScaleStep);
```

Comment:
again, the word temp isnt needed
and id just name the vectors by what theyre doing, velocity and acceleration